std/assembly/util/string.ts

```diff
@@ -369,23 +369,46 @@ export function joinFloatArray<T>(dataStart: usize, length: i32, separator: stri
   return result;
 }
 
-export function joinStringArray(dataStart: usize, length: i32, separator: string): string {
+export function joinStringArray<T>(dataStart: usize, length: i32, separator: string): string {
```

Comment:
Wondering if having a generic parameter here is worth it regarding code size. Is this significantly faster, justifying compiling two distinct Wasm functions?

Reply:
Reverted this